packages/block-library/src/navigation/index.php

```diff
@@ -245,10 +245,10 @@ function block_core_navigation_build_css_font_sizes( $attributes ) {
  * @return string
  */
 function block_core_navigation_render_submenu_icon() {
-	return '<svg xmlns="http://www.w3.org/2000/svg" width="12" height="12" viewBox="0 0 12 12" fill="none" aria-hidden="true" focusable="false"><path d="M1.50002 4L6.00002 8L10.5 4" stroke-width="1.5"></path></svg>';
+	$submenu_icon = '<svg xmlns="http://www.w3.org/2000/svg" width="12" height="12" viewBox="0 0 12 12" fill="none" aria-hidden="true" focusable="false"><path d="M1.50002 4L6.00002 8L10.5 4" stroke-width="1.5"></path></svg>';
+	return apply_filters( 'block_core_navigation_render_submenu_icon', $submenu_icon );
```

Comment:
Thanks for picking this up.

The problem with filtering here is that the user could have selected a particular icon in the editor, but then a filter applied by a plugin renders something completely different on the frontend, which results in a very confusing experience.

I think ideally the solution would be a filter that modifies the array of available icons for the user to select.

That may require JS and PHP code though.